xrpl-hooks-docs/md/hooks-guard-in-for.md

```diff
@@ -1,14 +1,56 @@
 # hooks-guard-in-for
 
-A guard is a marker that must be placed in your code at the top of each loop. Consider the following for-loop in C:
+Consider the following for-loop in C:
 
 ```c
-  #define GUARD(maxiter) _g(__LINE__, (maxiter)+1)
+#define GUARD(maxiter) _g(__LINE__, (maxiter)+1)
+for (int i = 0; GUARD(3), i < 3; ++i)
+```
+
+To satisfy the guard rule when using a for-loop in C guard should be 
+place either in the condition part of the loop, or as a first call in loop body, e.g.
 
-  for (int i = 0; GUARD(3), i < 3; ++i)
+```c
+for(int i = 0; i < 3; ++i) {
+  GUARD(3);
+}
 ```
 
-<BR/>
-This is the only way to satisfy the guard rule when using a for-loop in C.
+In case of nested loops, the guard limit value should be 
+multiplied by a number of iterations in each loop, e.g.
+
+```c
+for(int i = 0; GUARD(3), i < 3; ++i) {
+  for (int j = 0; GUARD(17), j < 5; ++j)
+}
+```
+
+```
+(most descendant loop iterations + 1) * (each parent loops iterations) - 1
+```
+
+This checker will also calculate the GUARD limit and show a hint with proposed value 
```

Comment:
generally, I think this is just too implementation-oriented - it should be describing the problem in the hook code, to a programmer who's just got some annoying warning and may not really know why they should care, not dwell on implementation details in a program they probably don't even know exists...